lib/src/participant/local.dart

```diff
@@ -310,13 +313,24 @@ class LocalParticipant extends Participant<LocalTrackPublication> {
     logger.fine('setSourceEnabled(source: $source, enabled: $enabled)');
     final publication = getTrackPublicationBySource(source);
     if (publication != null) {
+      var track = publication.track;
+      var stopOnMute = stopMicTrackOnMute && track is LocalAudioTrack;
       if (enabled) {
         await publication.unmute();
+        if (stopOnMute) {
+          await track.restartTrack();
+          await track.onTrackStarted();
+        }
       } else {
         if (source == TrackSource.screenShareVideo) {
           await unpublishTrack(publication.sid);
         } else {
           await publication.mute();
+          if (stopOnMute) {
+            await track.sender?.setTrack(null);
+            await track.stop();
+            await track.onTrackStopped();
```

Comment:
Does this work with bluetooth headsets? In browsers, it switches profile from communication to high-quality when the audio track is destroyed.

I think ideally we leave it in the same communication mode (as long as mic track has been acquired). would stopping the track alone make mic recording indicator disappear?

Reply:
I just tested it with airpods, it doesn't seem to work on the bluetooth headset, I tracked that the mode is set to AVAudioSessionCategoryPlayback, but the mic indicator light will still be on

Reply:
I have fixed this issue, key steps:
1, set category = playback
2, restart the playout in the audio device module

https://github.com/webrtc-sdk/webrtc/commit/654e4baa96899dd8741d393eafc387653d6ecad9#diff-96494536ec12249ed599120315bbc5df4ca4b7612c4098e5fdce65e72065328dR128-R136